project/SparkBuild.scala

```diff
@@ -51,6 +51,11 @@ object BuildCommons {
   // Root project.
   val spark = ProjectRef(buildLocation, "spark")
   val sparkHome = buildLocation
+
+  val testTempDir = s"$sparkHome/target/tmp"
+  if (!new File(testTempDir).isDirectory()) {
+    require(new File(testTempDir).mkdirs())
+  }
```

Comment:
The problem with doing it this way is that `buildLocation` is just using the current directory as the build root, which is not always a valid assumption.  We should instead remove that `val` and use `projectRoot.value` anyplace that we need the root spark directory.